ToolBox/AlgoSeekOptionsConverter/AlgoSeekOptionsConverter.cs

```diff
@@ -339,57 +349,33 @@ public void Clean(DateTime date)
+        ///          Source: https://documentation.help/SharpZip/ICSharpCode.SharpZipLib.BZip2.BZip2.html
+        /// </remarks>>
+        private static bool DecompressOpraFile(FileInfo compressedRawDatafile, FileInfo rawDatafile)
         {
-            bool timedOut = false;
-
-            Func<object, string> readStream = streamReader => ((StreamReader)streamReader).ReadToEnd();
-
-            using (var process = new Process())
+            var outcome = false;
+            using (FileStream fileToDecompressAsStream = compressedRawDatafile.OpenRead())
+            using (FileStream decompressedStream = File.Create(rawDatafile.FullName))
             {
-                process.StartInfo.UseShellExecute = false;
-                process.StartInfo.CreateNoWindow = true;
-                process.StartInfo.RedirectStandardError = true;
-                process.StartInfo.RedirectStandardOutput = true;
-                process.StartInfo.FileName = zipper;
-                process.StartInfo.Arguments = cmdArgs;
-
-                process.Start();
-
-                using (var processWaiter = Task.Factory.StartNew(() => process.WaitForExit(execTimeout * 1000)))
-                using (var outputReader = Task.Factory.StartNew(readStream, process.StandardOutput))
-                using (var errorReader = Task.Factory.StartNew(readStream, process.StandardError))
+                try
                 {
-                    bool waitResult = processWaiter.Result;
-
-                    if (!waitResult)
-                    {
-                        process.Kill();
-                        Log.Trace("7Zip Process Killed: " + cmdArgs);
-                    }
-
-                    Task.WaitAll(new Task[] { outputReader, errorReader }, execTimeout * 1000);
-
-                    if (!waitResult)
-                    {
-                        Log.Error("7Zip timed out: " + cmdArgs);
-                        throw new Exception("7z timed out");
-                    }
-                    else
-                    {
-                        if (process.ExitCode > 0)
-                        {
-                            Log.Error("7Zip Exited Unsuccessfully: " + cmdArgs);
-                            Log.Error("7zip message {0}", process.StandardError.ReadToEnd());
-                            throw new Exception("7z exited unsuccessfully");
-                        }
-                        else
-                        {
-                            Log.Trace("7Zip Exited Successfully: " + cmdArgs);
-                        }
-                    }
+                    BZip2.Decompress(fileToDecompressAsStream, decompressedStream, true);
+                    outcome = true;
+                }
+                catch (Exception ex)
+                {
+                    Log.Error($"AlgoSeekOptionsConverter.DecompressOpraFile({compressedRawDatafile.Name}, {rawDatafile.Name}): SharpzipLib.BZip2.Decompress returned error: " + ex.Message);
```

Comment:
_everytime_ we call `Log.Error` with an `Exception` in scope, we _must_ pass the exception object to the message, this ensures the full stack and all inner exceptions make it into the log. As written, the entire stack trace is lost.